core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -270,13 +271,19 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
                 node.getUseConnectorNodePartitioning());
 
         Expression remainingDecomposedPredicate;
-        if (connectorExpression.isEmpty() || remainingConnectorExpression.isEmpty() || remainingConnectorExpression.equals(connectorExpression)) {
+        if (remainingConnectorExpression.isEmpty() || remainingConnectorExpression.equals(connectorExpression)) {
             remainingDecomposedPredicate = decomposedPredicate.getRemainingExpression();
         }
         else {
             Map<String, Symbol> variableMappings = assignments.values().stream()
-                    .collect(Collectors.toMap(Symbol::getName, Function.identity()));
-            remainingDecomposedPredicate = ConnectorExpressionTranslator.translate(session, remainingConnectorExpression.get(), plannerContext, variableMappings, new LiteralEncoder(plannerContext));
+                                                              .collect(Collectors.toMap(Symbol::getName, Function.identity()));
+            Expression translatedExpression = ConnectorExpressionTranslator.translate(session, remainingConnectorExpression.get(), plannerContext, variableMappings, new LiteralEncoder(plannerContext));
+            if (connectorExpression.isEmpty()) {
+                remainingDecomposedPredicate = ExpressionUtils.combineConjuncts(plannerContext.getMetadata(), translatedExpression, decomposedPredicate.getRemainingExpression());
+            }
+            else {
+                remainingDecomposedPredicate = translatedExpression;
```

Comment:
The conversion 
```java
 Optional<ConnectorExpression> connectorExpression = new ConnectorExpressionTranslator.SqlToConnectorExpressionTranslator(session, remainingExpressionTypes, plannerContext)
                .process(decomposedPredicate.getRemainingExpression());
```

is lossy.

i.e. `decomposedPredicate.getRemainingExpression()` contains full information
but then `connectorExpression` contains partial information
`remainingConnectorExpression` contains part of `connectorExpression`, so again not a full information
so then `translatedExpression` cannot supersede `remainingDecomposedPredicate`

i think `SqlToConnectorExpressionTranslator` should work like a `DomainTranslator`, i..e it should spit out (a) translated result and (b) remaining expression

@martint do you have a suggestion how to handle this for now, so that `SqlToConnectorExpressionTranslator` change can be a follow up?

Reply:
> is lossy.

That should not be the case. The translation should either succeed and be equivalent to the original expression, or fail and return Optional.empty().

Reply:
Indeed @martint it's not a problem, as it's not lossy.
Although it should be. If a conjunct cannot be translated, we still should push down the others that can. cc @hashhar 
But that's something we can follow up on.